static/app/views/settings/organizationGeneralSettings/organizationSettingsForm.tsx

```diff
@@ -50,10 +50,10 @@ class OrganizationSettingsForm extends AsyncComponent<Props, State> {
         saveOnBlur
         allowUndo
         initialData={initialData}
-        onSubmitSuccess={(_resp, model) => {
+        onSubmitSuccess={(updated, _model) => {
           // Special case for slug, need to forward to new slug
           if (typeof onSave === 'function') {
-            onSave(initialData, model.initialData);
+            onSave(initialData, updated);
```

Comment:
Returning the response data which is an `Organization` seems more useful in this scenario and will let us handle redirects better.